src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/helpers/terraform/get_accounts.py

```diff
@@ -0,0 +1,131 @@
+import boto3
+import json
+import os
+import logging
+from paginator import paginator
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+
+MANAGEMENT_ACC_ID = os.environ["MANAGEMENT_ACCOUNT_ID"]
+if("TARGET_OUS" in os.environ):
+    OU_PATH = os.environ["TARGET_OUS"]
+sts = boto3.client('sts')
+
+def main():
+    
+    accounts = get_accounts()
+    with open('accounts.json', 'w') as outfile:
+        json.dump(accounts, outfile)
+
+    if("TARGET_OUS" in os.environ):
+        accounts_from_ous = get_accounts_from_ous()
+        with open('accounts_from_ous.json', 'w') as outfile:
+            json.dump(accounts_from_ous, outfile)
+
+
+def list_organizational_units_for_parent(parent_ou):
+    organizations = get_boto3_client('organizations', f'arn:aws:sts::{MANAGEMENT_ACC_ID}:role/OrganizationAccountAccessRole-readonly', 'getaccountID')
+    organizational_units = [
+        ou
+        for org_units in organizations.get_paginator("list_organizational_units_for_parent").paginate(ParentId=parent_ou)
+        for ou in org_units['OrganizationalUnits']
+    ]
+    return organizational_units
+
+def get_accounts():
+    # Return an array of objects like this: [{'AccountId':'xxx','Email':''}]
+    account_details = [] # [{'AccountId':'123','Email':''},{'AccountId':'456','Email':''}]
+    LOGGER.info(
+            "Management Account ID: %s",
+            MANAGEMENT_ACC_ID
+        )
+    # Assume a role into the management accounts role to get account ID's and emails
+    organizations = get_boto3_client('organizations', f'arn:aws:sts::{MANAGEMENT_ACC_ID}:role/OrganizationAccountAccessRole-readonly', 'getaccountID')
+    for account in paginator(organizations.list_accounts):
+        if account['Status'] == 'ACTIVE':
+            account_details.append({
+                'AccountId': account['Id'],
+                'Email': account['Email']
+            })
+    return account_details
+
+def get_accounts_from_ous():
+    parent_ou_id = None
+    account_list = []
+    organizations = get_boto3_client('organizations', f'arn:aws:sts::{MANAGEMENT_ACC_ID}:role/OrganizationAccountAccessRole-readonly', 'getaccountID')
+    # Read organization root id
+    root_ids = []
+    for id in paginator(organizations.list_roots):
+        root_ids.append({
+            'AccountId': id['Id']
+        })
```

Comment:
Can be rewritten to:

```suggestion
    root_ids = list(
        map(
            lambda root: {'AccountId': root['Id']]},
            paginator(organizations.list_roots)
        )
    )
```